### PR TITLE
[SPARK-15906][MLlib] Add complementary naive bayes algorithm

### PR DESCRIPTION
**Pull request by tilumi**
## What changes were proposed in this pull request?

Add `ComplementaryNaiveBayes.scala` in package `org.apache.spark.mllib.classification` in MLlib module
## How was this patch tested?

`build/sbt -Pyarn -Phadoop-2.6 -Phive -Phive-thriftserver "test-only 
org.apache.spark.mllib.classification.NaiveBayesSuite"`
`build/sbt -Pyarn -Phadoop-2.6 -Phive -Phive-thriftserver "test-only org.apache.spark.ml.classification.NaiveBayesSuite"`

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala [hunk 1]
```diff
@@ -446,8 +460,14 @@ object NaiveBayes {
   /** String name for Bernoulli model type. */
   private[spark] val Bernoulli: String = "bernoulli"
 
+  /** String name for complementary multinomial model type. */
+  private[spark] val Complementary_Multinomial: String = "complementary_multinomial"
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala]: I don't think these should be unrelated types. Complementary-ness is orthogonal to binary vs multinomial. The constant name is not conventional in any event.

Comment (tilumi) [inline mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala]: Thanks for the comment. So should I create a `ComplementaryNaiveBayes` class?

Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala]: No, but you can try to refactor the existing implementations to have some "complementary" config. It's possible that later these are exposed as separate types, but underneath, no need to be.

Comment (tilumi) [inline mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala]: OK, I got it. I would add another parameter in NaiveBayse for indicating it's comlementary or not

**Pull request closed without merging**